### PR TITLE
Fix incremental build failures

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Review by Forgind — APPROVED** [review]
I'm assuming you don't have any files that aren't copied in the list?

File: eng/targets/Imports.targets [hunk 1]
```diff
@@ -368,4 +368,12 @@
       <Reference Remove="@(Reference)" Condition=" '%(FileName)' == 'Microsoft.VisualBasic' or '%(FileName)' == 'Microsoft.VisualBasic.Core' " />
     </ItemGroup>
   </Target>
+
+  <Target Name="WorkaroundDotnetMsbuild7161"
```
Comment (jasonmalinowski) [inline eng/targets/Imports.targets:372]: Small request: stick a comment with the explicit bug URL.

Comment (sharwell) [reply eng/targets/Imports.targets:373]: ➡️ Done

**Comment by sharwell** [comment]
> I'm assuming you don't have any files that aren't copied in the list?

@Forgind I'm not sure I understand the question here

File: eng/targets/Imports.targets [hunk 2]
```diff
@@ -56,7 +56,7 @@
           Otherwise, default to 'true'.
   -->
   <PropertyGroup Condition="'$(RunAnalyzersDuringBuild)' == ''">
-    <RunAnalyzersDuringBuild Condition="'$(DesignTimeBuild)' != 'true' AND '$(ContinuousIntegrationBuild)' != 'true' AND !('$(RoslynEnforceCodeStyle)' == 'true' OR '$(RunCodeAnalysis)' == 'true')">false</RunAnalyzersDuringBuild>
+    <RunAnalyzersDuringBuild Condition="'$(RoslynEnforceCodeStyle)' != 'true' AND '$(RunCodeAnalysis)' != 'true'">false</RunAnalyzersDuringBuild>
```
Comment (mavasani) [inline eng/targets/Imports.targets:59]: Is this change required for this fix or is just a cleanup?

**Review by mavasani — APPROVED** [review]
LGTM. Thanks for fixing this Sam!

File: eng/targets/Imports.targets (hunk 2 above)
Comment (sharwell) [reply eng/targets/Imports.targets:59]: ➡️ Required for this fix

File: src/Setup/DevDivVsix/Directory.Build.targets [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
 <Project>
-  <Import Project="..\..\..\Directory.Build.targets" />
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.targets', '$(MSBuildThisFileDirectory)../'))" />
```
Comment (jaredpar) [inline src/Setup/DevDivVsix/Directory.Build.targets:3]: Why do we need this set of changes?

Comment (sharwell) [reply src/Setup/DevDivVsix/Directory.Build.targets:3]: This is not strictly required, but it unifies all cases where a nested **Directory.Build.targets** is provided to import **PackageProject.targets** (previously the `Import` line varied based on file location, so copy/paste was potentially problematic).

**Review by jaredpar — APPROVED** [review]

**Comment by Forgind** [comment]
> > I'm assuming you don't have any files that aren't copied in the list?
> 
> @Forgind I'm not sure I understand the question here

That wasn't very clear, and I actually don't remember what exactly I meant. I believe the bug BenVillalobos had been trying to fix was when a file is in the list of files to copy but shouldn't actually be copied for some other reason, so it's skipped. If we keep touching the CopyUpToDate marker, it can keep building. I think that was a somewhat unusual case? But worth verifying that building several times in a row doesn't rebuild the target.
